### PR TITLE
Integrate Jakarta Mail 2.0.0 and Jakarta Activation 2.0.0 into GlassFish

### LINKED ISSUE #22908: Incorporate Jakarta Activation 2.0.0 into GlassFish 6.0

### LINKED ISSUE #22914: Incorporate Jakarta Mail 2.0.0 into GlassFish 6.0

### PR DESCRIPTION
**Pull request by bshannon**
Passes quicklook.

I'm marking this "draft" to prevent premature integration.

Once this is integrated, the TCK will fail, until I update it as well.

### CODE REVIEW

**Comment by bshannon** [comment]
Would the assigned reviewers please review this PR?  Thanks.

**Review by smillidge — COMMENTED** [review]
What's the reason for the test deletions?

File: appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/CreateMailResource.java [hunk 1]
```diff
@@ -45,17 +45,17 @@
...
         @RestEndpoint(configBean=Resources.class,
                 opType=RestEndpoint.OpType.POST,
-                path="create-javamail-resource",
-                description="create-javamail-resource")
+                path="create-mail-resource",
+                description="create-mail-resource")
 })
 @org.glassfish.api.admin.ExecuteOn(RuntimeType.ALL)
-@Service(name="create-javamail-resource")
+@Service(name="create-mail-resource")
```
Comment (smillidge) [inline appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/CreateMailResource.java]: This is a rename of an asadmin command so will need documentation updates

File: appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/DeleteMailResource.java [hunk 2]
```diff
@@ -46,18 +46,18 @@
 @RestEndpoints({
         @RestEndpoint(configBean = Resources.class,
                 opType = RestEndpoint.OpType.DELETE,
-                path = "delete-javamail-resource",
-                description = "delete-javamail-resource")
+                path = "delete-mail-resource",
```
Comment (smillidge) [inline appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/DeleteMailResource.java]: again a rename of a command so all these will need documentation updates

**Review by smillidge — APPROVED** [review]

**Comment by bshannon** [comment]
> What's the reason for the test deletions?

I don't think there are any test deletions.  There should only be renamings to remove
javax or javamail from the path name.

What tests seemed to be deleted?

File: appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/CreateMailResource.java (hunk 1 above)
Comment (bshannon) [reply appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/CreateMailResource.java]: Yes, I believe I updated all the documentation in this repository.
Where do I find any other documentation that would need to be updated?

How are we collecting information for "release notes"?  Or are we not doing that?

**Comment by pzygielo** [comment]
> What tests seemed to be deleted?
```
D	appserver/resources/javamail/javamail-connector/src/test/java/org/glassfish/resources/javamail/admin/cli/CreateJavaMailResourceTest.java
D	appserver/resources/javamail/javamail-connector/src/test/java/org/glassfish/resources/javamail/admin/cli/DeleteJavaMailResourceTest.java
D	appserver/resources/javamail/javamail-connector/src/test/java/org/glassfish/resources/javamail/admin/cli/ListJavaMailResourcesTest.java
```

**Comment by SplotyCode** [comment]
I guess you renamed them but forgot to register them for git tracking via git ads

File: appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/CreateMailResource.java (hunk 1 above)
Comment (smillidge) [reply appserver/resources/mail/mail-connector/src/main/java/org/glassfish/resources/mail/admin/cli/CreateMailResource.java]: Ideally we would be disciplined and create an issue for our change and give it a milestone in GitHub as that is the only issue tracker we have. That way we can work out our release notes. I will raise this conversation to the list.

**Comment by bshannon** [comment]
> I guess you renamed them but forgot to register them for git tracking via git add

I used "git mv" to rename them.

**Comment by bshannon** [comment]
I think I've gotten all the review I'm going to get on this so I'm marking it ready to merge.
If no one else merges it by tomorrow, I'll do it.

**Review by hs536 — APPROVED** [review]

**Comment by pzygielo** [comment]
@bshannon Just to make sure - can you please provide location of `CreateJavaMailResourceTest.java` after your changes? I can't find it:
```
$ git status 
HEAD detached at 6e02d3f7
$ find appserver/resources/javamail -name CreateJavaMailResourceTest.java
(no output)
```

**Comment by bshannon** [comment]
> @bshannon Just to make sure - can you please provide location of `CreateJavaMailResourceTest.java` after your changes? I can't find it:

That's because it's now called CreateMailResourceTest.java, to mirror
the create-mail-resource command.  "JavaMail" was removed from all the
command and file names because it's now called "Jakarta Mail".

**Comment by pzygielo** [comment]
```
$ git status
HEAD detached at 6e02d3f7
$ find . -name  CreateMailResourceTest.java
(no output)
$ git diff --name-status 8d3bfb4e 6e02d3f7 | grep Test.java
D	appserver/resources/javamail/javamail-connector/src/test/java/org/glassfish/resources/javamail/admin/cli/CreateJavaMailResourceTest.java
D	appserver/resources/javamail/javamail-connector/src/test/java/org/glassfish/resources/javamail/admin/cli/DeleteJavaMailResourceTest.java
D	appserver/resources/javamail/javamail-connector/src/test/java/org/glassfish/resources/javamail/admin/cli/ListJavaMailResourcesTest.java
$
```

So - can you please tell where I can find `CreateMailResourceTest.java` then?

**Comment by bshannon** [comment]
I'm pretty sure I used "git mv" to rename these files, but for some reason these files
ended up untracked.  I added them and pushed them.

Thanks for noticing!

**Pull request merged**